onmt/modules/LSTM.lua

```diff
@@ -104,8 +116,22 @@ function LSTM:_buildLayer(inputSize, hiddenSize)
   local x = inputs[3]
 
   -- Evaluate the input sums at once for efficiency.
-  local i2h = nn.Linear(inputSize, 4 * hiddenSize)(x)
-  local h2h = nn.Linear(hiddenSize, 4 * hiddenSize)(prevH)
+  local i2h = nn.Linear(inputSize, 4 * hiddenSize, false)(x)
```

Comment:
So first, in the non-batch-normalized case, by having both modules have biases, there's an unnecessary redundancy since they will just be added anyway.

In the batch normalized case, the authors of this [paper](https://arxiv.org/pdf/1603.09025v4.pdf) recommended batch normalizing the output before adding the bias, and having that be separate (see equation 6).

That all being said, I'll measure the differences and report back.